### PR TITLE
Handle implicit types

### PR DESCRIPTION
**Pull request by JeremyKuhne**
- Handle types that were implicitly loaded from System.dll
- Clean up security related methods in exception
- Some code cleanliness
- Add unit tests for util classes

### CODE REVIEW

**Comment by JeremyKuhne** [comment]
@dotnet-bot test Innerloop OSX Debug Build and Test (segfault in X509Certificates.Tests)

**Comment by JeremyKuhne** [comment]
@dotnet-bot test Innerloop OSX Release Build and Test (segfault in X509Certificates.Tests)

File: src/System.Configuration/src/System/Configuration/TypeUtil.cs [hunk 1]
```diff
@@ -9,25 +9,65 @@ namespace System.Configuration
...
+        private static string[] s_implicitAssemblies =
         {
-            Type type = null;
+            "System.Runtime",
+            "System.Runtime.Extensions",
+            "System.Collections",
+            "System.Collections.Concurrent",
+            "System.Collections.Specialized",
+
+            // Historically this is the assembly we would look in
```
Comment (ericstj) [inline src/System.Configuration/src/System/Configuration/TypeUtil.cs]: Let's say, this code used to live in System on desktop, so an unqualified type name would have looked first in System then in mscorlib.

Given that, I think this list should start with "System", then list "mscorlib".

All this is based on some cursory tests of a desktop console app.  We should probably take a close look at what desktop did and make sure that this is a correct model.

Now that we're bringing back all the classic assemblies as a *part of the framework* I wonder if we could have reflection do the mscorlib check.  @weshaggard what do you think?

**Comment by jkotas** [comment]
The plan I have seen for System.Configuration was to enable it for user libraries, but not for the configuration of the framework itself.

Is it still the plan? 

If it is, I do not think any of this compat with Everett goo should be needed. And even if it was, I still think we should drop it for simplicity because of it is going to have number of cracks anyway. Bug-for-bug compatibility with Everett should be non-goal.

**Comment by JeremyKuhne** [comment]
> The plan I have seen for System.Configuration was to enable it for user libraries, but not for the configuration of the framework itself.

Yes, that is the current plan.

> Bug-for-bug compatibility with Everett should be non-goal.

In general I agree. In this specific case I'm only looking to be able to instantiate common types that came from System, such as Uri and some of the collections. I think it is pretty likely people were relying on it for some of their data given that we've always supported it.  I don't care if we're exact, only that we get common cases.

**Pull request merged**